### PR TITLE
Send delivery feedback for a message

### PR DESCRIPTION
**Pull request by HannaTul**

### CODE REVIEW

File: src/main/java/com/sinch/xms/ApiConnection.java [hunk 1]
```diff
@@ -1145,7 +1181,7 @@ public Tags replaceTags(BatchId id, Tags tags) throws InterruptedException, ApiE
 	 * @throws InterruptedException if the current thread was interrupted while
 	 *                              waiting
 	 * @throws ApiException         if an error occurred while communicating with
-	 *                              XMS
+	 *                              XMSIncoming SMS
```
Comment (miksta-sinch) [inline src/main/java/com/sinch/xms/ApiConnection.java:1184]: Why this change?

File: src/main/java/com/sinch/xms/api/MtBatchDeliveryFeedbackResult.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.immutables.value.Value;
+
+import javax.annotation.Nonnull;
+
+@Value.Immutable
+@ValueStylePackage
+@JsonDeserialize(builder = MtBatchDeliveryFeedbackResult.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+abstract public class MtBatchDeliveryFeedbackResult {
```
Comment (miksta-sinch) [inline src/main/java/com/sinch/xms/api/MtBatchDeliveryFeedbackResult.java:13]: Cannot we just use Void instead?

File: src/main/java/com/sinch/xms/api/MtBatchSmsCreate.java [hunk 3]
```diff
@@ -100,6 +100,18 @@
 	@JsonProperty("callback_url")
 	public abstract URI callbackUrl();
 
+	/**
+	 * The URL to which batch callbacks should be sent. If <code>null</code> then
```
Comment (miksta-sinch) [inline src/main/java/com/sinch/xms/api/MtBatchSmsCreate.java:104]: feedback_enabled is not the URL

File: src/main/java/com/sinch/xms/SinchSMSApi.java [hunk 4]
```diff
@@ -78,6 +79,21 @@
...
+	 *
+	 * @return a builder of delivery feedback
+	 */
+	@Nonnull
+	public static FeedbackDeliveryCreate.Builder deliveryFeedback() {
+		return FeedbackDeliveryCreate.builder();
+	}
+
+	/**
+	 * Returns a freshly created builder for text message updates.
```
Comment (thomasf147) [inline src/main/java/com/sinch/xms/SinchSMSApi.java:93]: This should not be here right?

File: src/main/java/com/sinch/xms/api/FeedbackDeliveryCreate.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.immutables.value.Value;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+@Value.Immutable
+@ValueStylePackage
+@JsonDeserialize(builder = FeedbackDeliveryCreate.Builder.class)
+@JsonTypeName("delivery_feedback")
```
Comment (thomasf147) [inline src/main/java/com/sinch/xms/api/FeedbackDeliveryCreate.java:13]: There is no subtype for this so this line can be removed

File: src/test/java/com/sinch/xms/ApiConnectionIT.java [hunk 6]
```diff
@@ -1628,6 +1630,35 @@ public void completed(RecipientDeliveryReport result) {
...
+
+		FeedbackDeliveryCreate request =
+		        SinchSMSApi.deliveryFeedback()
+						.addRecipient("+15551231234","+15551256344")
+				.build();
+
+		MtBatchDeliveryFeedbackResult expected = MtBatchDeliveryFeedbackResult.builder()
+				.build();
+
+		stubPostResponse(expected, path, 200);
```
Comment (thomasf147) [inline src/test/java/com/sinch/xms/ApiConnectionIT.java:1648]: The response should actually be 202 and not 200.

File: src/main/java/com/sinch/xms/ApiConnection.java [hunk 7]
```diff
@@ -1051,6 +1058,44 @@ public RecipientDeliveryReport fetchDeliveryReport(BatchId id, String recipient)
...
+	 * 	 * @param callback  a callback that is activated at call completion
+	 */
+	public void createDeliveryFeedbackAsync(BatchId id, FeedbackDeliveryCreate feedbackDeliveryCreate,
+																			FutureCallback<EmptyResult> callback) {
+
+		HttpPost req = post(batchDeliveryFeedbackEndpoint(id), feedbackDeliveryCreate);
+
+		HttpAsyncRequestProducer producer = new BasicAsyncRequestProducer(endpointHost(), req);
+
+		HttpAsyncResponseConsumer<EmptyResult> consumer = jsonAsyncConsumer(EmptyResult.class);
```
[the remark this answers is not in the document]
Comment (thomasf147) [reply src/main/java/com/sinch/xms/ApiConnection.java:1094]: Looks like the consumer does not appreciate Void

File: src/main/java/com/sinch/xms/api/MtBatchSmsCreate.java [hunk 8]
```diff
@@ -97,6 +97,15 @@
   @JsonProperty("callback_url")
   public abstract URI callbackUrl();
 
+  /**
+   * The boolean value which determines if feedback is allowed to sent
```
Comment (thomasf147) [inline src/main/java/com/sinch/xms/api/MtBatchSmsCreate.java:101]: allowed to be sent
